### PR TITLE
Read nft by creator

### PR DESCRIPTION
**Pull request by rex-god**
- Read NFT by `creator_address` (with pagination from DB & Blockchain)
- Search NFT by attributes (`eq`, `gt`, `gte`, `lt`, `lte` operation), creators, wallet (required query), network. (with pagination - currently only fetch data from DB)

### CODE REVIEW

File: src/modules/nft/components/read-nft/read-nft.service.ts [hunk 1]
```diff
@@ -49,6 +49,38 @@ export class ReadNftService {
...
+
+        return dbNftInfo.map((nft) => {
+          return getNftDbResponseFromNftInfo(nft);
+        });
+      } else {
+        //not available in DB, fetch from blockchain
+        const fetchAllNft = new FetchAllNftByCreatorDto(network, creator_address);
+        const chainNfts = await this.remoteDataFetcher.fetchAllNftsByCreator(fetchAllNft);
+        const nftInfo = chainNfts.map((nft) => nft.getNftInfoDto());
+        const nftReadByCreatorEvent = new NftReadByCreatorEvent(creator_address, network, nftInfo);
```
Comment (codinpsycho) [inline src/modules/nft/components/read-nft/read-nft.service.ts:72]: Why is there a need for 2 read events?

Comment (rex-god) [reply src/modules/nft/components/read-nft/read-nft.service.ts:72]: `NftReadByCreatorEvent` saves all data on DB.

File: src/modules/nft/components/search-nft/search-nft.service.ts [hunk 2]
```diff
@@ -30,4 +32,55 @@ export class SearchNftService {
 
     return result;
   }
+
+  async searchNfts(searchNftDto: SearchNftsDto): Promise<any> {
```
Comment (vgvishesh) [inline src/modules/nft/components/search-nft/search-nft.service.ts:36]: write unit test cases to test this function properly.

File: src/modules/nft/components/read-nft/read-nft.service.ts [hunk 3]
```diff
@@ -49,6 +49,38 @@ export class ReadNftService {
...
+        this.eventEmitter.emit('all.nfts.read.by.creator', nftReadByCreatorEvent);
+
+        const nfts = chainNfts.map((nft) => nft.getNftDbResponse());
+        return nfts;
+      }
+    } catch (error) {
+      console.log(error);
+      throw new HttpException(error.message, error.status);
```
Comment (vgvishesh) [inline src/modules/nft/components/read-nft/read-nft.service.ts:80]: use newProgramError to log and throw run time exceptions. Checkout the latest code on main branch. Don't throw httpException from service layers.

**Pull request closed without merging**